force-app/main/default/classes/CourseRegistrationHandler.cls

```diff
@@ -0,0 +1,135 @@
+global class CourseRegistrationHandler extends MyTriggers {
+    global override void onAfterDelete() {
+        getCourseIdToReduce((List<CourseRegistration__c>) records);
+    }
+
+    global override void onAfterUndelete() {
+        getCourseIdToIncrease((List<CourseRegistration__c>) records);
+    }
+
+    global override void onAfterInsert() {
+        getCourseIdToIncrease((List<CourseRegistration__c>) records);
+    }
+
+    global override void onAfterUpdate(Map<Id, sObject> triggerOldMap) {
+        List<String> fieldsToCheck = new List<String>{'Status__c'};
+        Id courseId;
+        for (CourseRegistration__c cr : (List<CourseRegistration__c>) records) {
+            if(cr.Status__c == 'Påmeldt' && 
+               MyTriggers.hasChangedFields(fieldsToCheck, cr, triggerOldMap.get(cr.Id))) 
+               {
+                courseId = cr.Course__c;
+                increaseCourseNumParticipants(courseId);
+                if(triggerOldMap.get(cr.Id).get('Status__c') == 'Venteliste'){
+                    courseId = cr.Course__c;
+                    reduceCourseWaitinglist(courseId);
+                    }
+               }
+            else if(MyTriggers.hasChangedFields(fieldsToCheck, cr, triggerOldMap.get(cr.Id)) &&
+                    triggerOldMap.get(cr.Id).get('Status__c') == 'Påmeldt')
+                    {
+                     courseId = cr.Course__c;
+                     reduceCourseNumParticipants(courseId);
+                     if(cr.Status__c == 'Avmeldt' &&
+                        checkWaitinglist(cr.Course__c) == true)
+                        {
+                            courseId = cr.Course__c;
+                            setNextInLineToRegistered(courseId);
```

Comment:
```suggestion
                            setNextInLineToRegistered(cr.Course__c);
```

Bedre å bare sende inne `cr.Course__c` fordi da slipper du 10 linjer med `courseId = cr.Course__c;`. Kan eventuelt putte `Id courseId = cr.Course__c;` på toppen (etter if) og referere til den hver gang. Jeg ville selv gått for `cr.Course__c`.